mlir/include/Gradient/IR/GradientOps.td

```diff
@@ -88,28 +88,33 @@ def AdjointOp : Gradient_Op<"adjoint", [AttrSizedOperandSegments]> {
     }];
 }
 
-def BackpropOp : Gradient_Op<"backprop", [AttrSizedOperandSegments]> {
+def BackpropOp : Gradient_Op<"backprop", [AttrSizedOperandSegments,
+        DeclareOpInterfaceMethods<SymbolUserOpInterface>]> {
     let summary = "Perform classic automatic differentiation using Enzyme AD.";
 
     let arguments = (ins
         FlatSymbolRefAttr:$callee,
         Variadic<AnyType>:$args,
-        Variadic<AnyType>:$quantum_jacobian,
-        Variadic<AnyMemRef>:$data_in,
+        Variadic<MemRefOf<[AnyFloat]>>:$diffArgShadows,
+        Variadic<MemRefOf<[AnyFloat]>>:$calleeResults,
```

Comment:
Why did we need the callee results here?

Reply:
The idea was to handle all allocations during the bufferization stage. At the moment the buffers for the callee results (required by Enzyme) were allocated during the lowering, but I think it makes more sense during the bufferization stage.

Reply:
Enzyme requires allocated space for the primal (callee) results even though we don't need their values. Adding them as arguments here allows us to allocate this space during bufferization, which feels more appropriate than allocating them during conversion to the LLVM dialect.

Reply:
Thanks a lot :+1: